autogpt/core/llm/base.py

```diff
@@ -1,5 +1,145 @@
 import abc
+import typing
+from dataclasses import dataclass, field
+from typing import Dict, List
 
+if typing.TYPE_CHECKING:
+    from autogpt.core.configuration.base import Configuration
+    from autogpt.core.logging.base import Logger
+    from autogpt.core.planning.base import ModelPrompt
+    from autogpt.core.plugin.base import PluginManager
+    from autogpt.core.workspace.base import Workspace
+
+
+@dataclass
+class ModelInfo:
+    """Struct for model information.
+
+    Would be lovely to eventually get this directly from APIs, but needs to be scraped from
+    websites for now.
+
+    """
+
+    name: str
+    prompt_token_cost: float
+    completion_token_cost: float
+    max_tokens: int
+
+
+@dataclass
+class ChatModelInfo(ModelInfo):
+    """Struct for chat model information."""
 
-class LanguageModel(abc.ABC):
     pass
+
+
+@dataclass
+class EmbeddingModelInfo(ModelInfo):
+    """Struct for embedding model information."""
+
+    embedding_dimensions: int
+
+
+@dataclass
+class LLMResponse:
+    """Standard response struct for a response from an LLM model."""
+
+    model_info: ModelInfo
+    prompt_tokens_used: int = 0
+    completion_tokens_used: int = 0
+
+
+@dataclass
+class EmbeddingModelResponse(LLMResponse):
+    """Standard response struct for a response from an embedding model."""
+
+    embedding: List[float] = field(default_factory=list)
+
+    def __post_init__(self):
+        if self.completion_tokens_used:
+            raise ValueError("Embeddings should not have completion tokens used.")
+
+
+@dataclass
+class ChatModelResponse(LLMResponse):
+    """Standard response struct for a response from an LLM model."""
+
+    content: str = None
+
+
+class LanguageModel(abc.ABC):
+    configuration_defaults = {"language_model": {}}
+
+    @abc.abstractmethod
+    def __init__(
+        self,
+        configuration: Configuration,
```

Comment:
Just wondering, why are we importing the entire configuration here?

Reply:
I'm starting permissive because we're going to have to port the existing logic into this system and we're going to find edge cases in that port that we can't isolate without significant work.  We don't even have the most basic form of encapsulation yet.  We can tighten contracts later once we have a little structure to lean on, and our priority in this work is getting that structure in quickly.